src/components/include/utils/logger.h

```diff
@@ -33,151 +33,110 @@
-// it's need, for example, when crash happend
-#define FLUSH_LOGGER() logger::flush_logger()
-
-#define LOG4CXX_IS_TRACE_ENABLED(logger) logger->isTraceEnabled()
-
-log4cxx_time_t time_now();
-
-#define LOG_WITH_LEVEL(loggerPtr, logLevel, logEvent)                \
-  do {                                                               \
-    if (logger::logs_enabled()) {                                    \
-      if (logger::logger_status != logger::DeletingLoggerThread) {   \
-        if (loggerPtr->isEnabledFor(logLevel)) {                     \
-          std::stringstream accumulator;                             \
-          accumulator << logEvent;                                   \
-          logger::push_log(                                          \
-              loggerPtr,                                             \
-              logLevel,                                              \
-              accumulator.str(),                                     \
-              time_now(),                                            \
-              LOG4CXX_LOCATION,                                      \
-              ::log4cxx::spi::LoggingEvent::getCurrentThreadName()); \
-        }                                                            \
-      }                                                              \
-    }                                                                \
+#define SDL_DEINIT_LOGGER() logger::Logger::instance().DeInit();
+
+#define LOG_WITH_LEVEL(logLevel, logEvent)                             \
+  do {                                                                 \
+      if (logger::logger_status != logger::DeletingLoggerThread) {     \
+        if (logger::Logger::instance().IsEnabledFor(   \
+                logger_, logLevel)) {                                  \
+          std::stringstream accumulator;                               \
+          accumulator << logEvent;                                     \
+          logger::LogMessage message{                               \
+              logger_,                                                 \
+              logLevel,                                                \
+              accumulator.str(),                                       \
+              std::chrono::high_resolution_clock::now(),               \
+              logger::LocationInfo{                                 \
+                  __FILE__, __PRETTY_FUNCTION__, __LINE__},            \
+              std::this_thread::get_id()};                             \
+          logger::Logger::instance().PushLog(message); \
+        }                                                              \
+      }                                                                \
   } while (false)
 
-#undef LOG4CXX_TRACE
-#define LOG4CXX_TRACE(loggerPtr, logEvent) \
-  LOG_WITH_LEVEL(loggerPtr, ::log4cxx::Level::getTrace(), logEvent)
+#define SDL_TRACE(logEvent) \
```

Comment:
@VladSemenyuk I believe that logger macroses should contain at least word "log" to understand what is their purpose.
So I would recommend to use the following naming:
`SDL_LOG_TRACE`
`SDL_LOG_DEBUG`
`SDL_LOG_INFO`
`SDL_LOG_WARN`
`SDL_LOG_ERROR`
`SDL_LOG_FATAL`
`SDL_LOG_WARN_WITH_ERRNO`
`SDL_LOG_ERROR_WITH_ERRNO`